pyscf_ipu/nanoDFT/nanoDFT.py

```diff
@@ -199,10 +199,18 @@ def init_dft_tensors_cpu(mol, opts, DIIS_iters=9):
     grid_weights    = grids.weights                                 # (grid_size,) = (45624,) for C6H6
     coord_str       = 'GTOval_cart_deriv1' if mol.cart else 'GTOval_sph_deriv1'
     grid_AO         = mol.eval_gto(coord_str, grids.coords, 4)      # (4, grid_size, N) = (4, 45624, 9) for C6H6.
-    if opts.ao_threshold is not None:
+    if opts.ao_threshold > 0.0:
```

Comment:
Looks good!

Reply:
Really like the simplicity: we just pre-process remove and only change grid_size so exchange_correlation code remains the same!